cargo-insta/src/inline.rs

```diff
@@ -47,10 +48,17 @@ impl FilePatcher {
     }
 
     pub fn save(&self) -> Result<(), Box<dyn Error>> {
-        let mut f = fs::File::create(&self.filename)?;
+        // We use a temp file and then atomically rename to prevent a
+        // file watcher restarting the process midway through the write.
+        let mut temp_file = NamedTempFile::new()?;
```

Comment:
This would need to be `new_in` with the target folder. Atomic renames are not possible from different file systems and there is no guarantee that the picked temporary path is on the same drive.

Example: https://github.com/mitsuhiko/self-replace/blob/61cd85f6c1fe8af98bc8c17e3d993f87253dc986/src/unix.rs#L23-L30

Reply:
Thanks, done!

Tests pass (though I haven't done a full manual test, lmk if I should)